src/main/java/pl/tomaszdziurko/jvm_bloggers/blogs/BloggersDataUpdater.java

```diff
@@ -55,22 +53,28 @@ protected void updateSingleEntry(BloggerEntry bloggerEntry, UpdateSummary update
     private void updateBloggerIfThereAreSomeChanges(BloggerEntry bloggerEntry,
                                                     UpdateSummary updateSummary,
                                                     Blog existingBlogger) {
-        bloggerEntry.setUrl(
-            syndFeedFactory.validUrlFromRss(bloggerEntry.getRss()).orElse(existingBlogger.getUrl())
-        );
-        
-        if (!isEqual(existingBlogger, bloggerEntry)) {
+
+        Optional<String> validBlogUrl = extractValidBlogUrlFromFeed(bloggerEntry.getRss());
+        validBlogUrl.ifPresent(bloggerEntry::setUrl);
+
+        if (somethingChangedInBloggerData(existingBlogger, bloggerEntry)) {
```

Comment:
\[Checkstyle\] WARNING: 'if' have incorrect indentation level 8, expected level should be 4\.